CMakeLists.txt

```diff
@@ -72,13 +82,10 @@ ELSE()
   ELSE()
     ADD_COMPILE_FLAG("-fPIC")
   ENDIF()
-  IF(uppercase_CMAKE_BUILD_TYPE STREQUAL "DEBUG")
-    ADD_COMPILE_FLAG("-O0")
-    ADD_COMPILE_FLAG("-g3")
-  ELSE()
-    ADD_COMPILE_FLAG("-O2")
-    ADD_DEFINITIONS("-UNDEBUG")  # Keep asserts.
-  ENDIF()
+  ADD_DEBUG_COMPILE_FLAG("-O0")
+  ADD_DEBUG_COMPILE_FLAG("-g3")
+  ADD_NONDEBUG_COMPILE_FLAG("-O2")
```

Comment:
Yeah, `-O3`/`-g` are CMake defaults. Considered deleting these directives altogether, but thought that there was some specific desire to do just `-O2 and -g3` and wanted to keep the behavior identical. (Fixing platform specific bugs doesn't generally mix well with changing behavior)

I would like to remove the `-UNDEBUG` directive altogether and have people build the CMake `RelWithDebInfo` target when they want assertions, so that `Release` could be without assertions for performance purposes. Currently `Release` comes with assertions enabled so not nice for performance. But that change can occur separately if agreed on.